rest/src/main/kotlin/builder/interaction/FollowupMessageBuilders.kt

```diff
@@ -207,46 +129,73 @@ class PublicFollowupMessageCreateBuilder : RequestBuilder<MultipartFollowupMessa
-     * (ping everything), calling this function but not configuring it before the request is build will result in all
-     * pings being ignored.
-     */
-    @OptIn(ExperimentalContracts::class)
-    inline fun allowedMentions(block: AllowedMentionsBuilder.() -> Unit = {}) {
-        contract { callsInPlace(block, InvocationKind.EXACTLY_ONCE) }
-        allowedMentions = (allowedMentions ?: AllowedMentionsBuilder()).apply(block)
-    }
+    override var allowedMentions: AllowedMentionsBuilder? by ::_allowedMentions.delegate()
 
-    @OptIn(ExperimentalContracts::class)
-    @KordPreview
-    inline fun actionRow(builder: ActionRowBuilder.() -> Unit) {
-        contract {
-            callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
-        }
+    override val embeds: MutableList<EmbedBuilder> = mutableListOf()
+
+    override val components: MutableList<MessageComponentBuilder> = mutableListOf()
 
-        components.add(ActionRowBuilder().apply(builder))
-    }
 
     override fun toRequest(): MultipartFollowupMessageCreateRequest =
         MultipartFollowupMessageCreateRequest(
             FollowupMessageCreateRequest(
-                content = Optional.Value(content),
+                content = _content,
+                embeds = Optional.missingOnEmpty(embeds.map(EmbedBuilder::toRequest)),
                 tts = _tts,
                 allowedMentions = _allowedMentions.map { it.build() },
                 components = Optional.missingOnEmpty(components.map(MessageComponentBuilder::build))
             ),
         )
 
+
+}
+sealed interface BaseFollowupMessageBuilder<T> : RequestBuilder<T> {
+
+    var tts: Boolean?
+
+    var allowedMentions: AllowedMentionsBuilder?
+
+    val embeds: MutableList<EmbedBuilder>
+
+    val components: MutableList<MessageComponentBuilder>
+
+    val content: String?
+
+}
+
+@OptIn(ExperimentalContracts::class)
+@KordPreview
+inline fun <T> BaseFollowupMessageBuilder<T>.actionRow(builder: ActionRowBuilder.() -> Unit) {
+    contract {
+        callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
+    }
+
+    components.add(ActionRowBuilder().apply(builder))
+}
+
+@OptIn(ExperimentalContracts::class)
+inline fun <T> BaseFollowupMessageBuilder<T>.embed(builder: EmbedBuilder.() -> Unit) {
+    contract { callsInPlace(builder, InvocationKind.EXACTLY_ONCE) }
+    embeds += EmbedBuilder().apply(builder)
 }
+
+/**
+ * Configures the mentions that should trigger a mention (aka ping). Not calling this function will result in the default behavior
+ * (ping everything), calling this function but not configuring it before the request is build will result in all
+ * pings being ignored.
+ */
+@OptIn(ExperimentalContracts::class)
+inline fun <T> BaseFollowupMessageBuilder<T>.allowedMentions(block: AllowedMentionsBuilder.() -> Unit = {}) {
+    contract { callsInPlace(block, InvocationKind.EXACTLY_ONCE) }
+    allowedMentions = (allowedMentions ?: AllowedMentionsBuilder()).apply(block)
+}
```

Comment:
This should be declared at the top of the file instead for more intuitive reading.